src/schema/formats.js

```diff
@@ -1,48 +1,32 @@
 import { URL } from 'whatwg-url';
 
-const VALIDNUMRX = /^[0-9]{1,5}$/;
-
 // Firefox's version format is laxer than Chrome's, it accepts:
 // https://developer.mozilla.org/en-US/docs/Toolkit_version_format
 // We choose a slightly restricted version of that format (but still more
 // permissive than Chrome) to allow Beta addons, per:
 // https://developer.mozilla.org/en-US/Add-ons/AMO/Policy/Maintenance
-const TOOLKIT_VERSION_REGEX = /^(\d+\.?){1,3}\.(\d+([A-z]+(-?[\dA-z]+)?))$/;
+const VERSION_PART =
+  '(?:0|[1-9]\\d{0,3}|[1-5]\\d{4}|6(?:[0-4]\\d{3}|5(?:[0-4]\\d{2}|5(?:[0-2]\\d|3[0-5]))))';
+const BETA_PART = '(?:a(?:lpha)?|b(?:eta)?|pre|rc)\\d*';
+const VERSION_REGEXP =
+  new RegExp(`^${VERSION_PART}(?:\\.${VERSION_PART}){0,3}(?:${BETA_PART})?$`);
+const TOOLKIT_REGEXP =
+  new RegExp(`^${VERSION_PART}(?:\\.${VERSION_PART}){0,3}${BETA_PART}$`);
 
 export function isValidVersionString(version) {
   // We should be starting with a string.
   if (typeof version !== 'string') {
     return false;
   }
-  // If valid toolkit version string, return true early
-  if (TOOLKIT_VERSION_REGEX.test(version)) {
-    return true;
-  }
-  const parts = version.split('.');
-  if (parts.length > 4) {
-    return false;
-  }
-
-  for (let i = 0; i < parts.length; i++) {
-    let part = parts[i];
-    // Leading or multiple zeros not allowed.
-    if (part.startsWith('0') && part.length > 1) {
-      return false;
-    }
-    // Disallow things like 123e5 which parseInt will convert.
```

Comment:
This warrants a test, can you add that?

Reply:
Done.
And also added more test patterns.

Reply:
Ah ` '123e5'` in the invalid patterns, somehow I missed that.

Nice, thanks!